### DISCUSSION TITLE
RestDocsTest에 모든 Service 객체를 정의하는 것에 대해

### DISCUSSION THREAD

**Discussion by amaran-th** [BDD-CLUB/01-doo-re-back#48] [논의 사항]
현재 보름님께서 작성하신 RestDocsTest 관련 코드에 따르면, 모킹 대상인 서비스 객체가 각각의 API 테스트 클래스에 정의되어 있습니다.
이 방식의 경우, 추후 다른 api 테스트이 작성됐을 때 테스트 클래스마다 다른 컨텍스트를 갖게 되어 성능이 느려질 수 있습니다.
제 경험상 프로젝트가 커졌을 때 테스트마다 매번 새로운 컨텍스트를 띄우는 경우, 지연되는 시간이 체감될 정도였습니다.

그래서 제가 제안하고 싶은 방식은, 컨텍스트를 통일시키기 위해 RestDocsTest 클래스에 모든 Service 객체를 정의하는 것입니다.
보기엔 다소 불편해 보일 수 있지만, 성능 향상을 위한 방안이니 긍정적으로 생각해주시면 감사하겠습니다.
```java
@Import(RestDocsConfiguration.class)
@ExtendWith(RestDocumentationExtension.class)
public abstract class RestDocsTest {

    @MockBean
    protected LoginService loginService;

    @MockBean
    protected MemberCommandService memberCommandService;
    ...
```
문제 이해를 위해 [참고할 수 있는 게시글](https://mangkyu.tistory.com/244) 남깁니다!

**Comment by shkisme** [comment]
저는 `RestDocsTest`의 깃 충돌 문제를 고려해서 이렇게 코드를 작성한 기억이 나는데, 매번 새로운 컨텍스트가 생성되는 줄 몰랐네요;; 충분히 고려해 볼 사항인 것 같습니다. 👍

**Comment by lcqff** [comment]
글 읽어보고 왔는데 이해한 거 같습니다.
단 정의될 객체가 service만 있는 건 아니게 될 거 같아서?? 코드가 길어질까 걱정은 되긴하네요 😅
아니면 정의될 객체만 모아두는 코드를 따로 작성해서 RestDocsTest에 상속해줄 순 없을까요?

**Reply by lcqff** [reply]
근데 상속 얘기는 이후 얘기될 디스커션 때문에 뺄게요~~ 저는 좋습니다

**Comment by JJimini** [comment]
좋은 것 같습니다!!

**Comment by 02ggang9** [comment]
좋은 것 같습니다~

**Comment by lcqff** [comment]
땅땅땅

**Comment by 02ggang9** [comment]
성능을 생각해 Discussion 내용대로 Service 객체를 정의하도록 결정되었습니다~
